packages/runtime/src/state/actions.ts

```diff
@@ -136,6 +161,7 @@ type RegisterBuilderComponentAction = {
     meta: ComponentMeta
     serializedControls: Record<string, SerializedControl>
   }
+  transferables?: Transferable[]
```

Comment:
Add optional `transferables` prop so we can dispatch `REGISTER_BUILDER_COMPONENT` from the middleware instead of repurposing `REGISTER_COMPONENT` -- see below.